samza-core/src/test/java/org/apache/samza/task/TestStreamOperatorTask.java

```diff
@@ -80,4 +84,34 @@ public void testExceptionsInProcessInvokesTaskCallback() throws InterruptedExcep
         mockTaskCoordinator, mockTaskCallback);
     failureLatch.await();
   }
+
+  /**
+   * Tests if the appropriate SamzaException is propagated to the TaskCallback if there is no InputOperator for a given
+   * SystemStream in the OperatorGraph.
+   * */
+  @Test
+  public void testExceptionIfInputOperatorMissing() throws NoSuchFieldException, IllegalAccessException {
+    IncomingMessageEnvelope mockIme = mock(IncomingMessageEnvelope.class, RETURNS_DEEP_STUBS);
+    SystemStream testSystemStream = new SystemStream("foo", "bar");
+    when(mockIme.getSystemStreamPartition().getSystemStream()).thenReturn(testSystemStream);
+
+    OperatorImplGraph mockOperatorImplGraph = mock(OperatorImplGraph.class);
+    when(mockOperatorImplGraph.getInputOperator(anyObject())).thenReturn(null);
+    // OperatorImplGraph is a private field ('operatorImplGraph') in StreamOperatorTask and need to be set with
+    // the mocked value reflectively
+    StreamOperatorTask operatorTask = new StreamOperatorTask(mock(OperatorSpecGraph.class));
+    Field operatorImplGraphField = operatorTask.getClass().getDeclaredField("operatorImplGraph");
+    operatorImplGraphField.setAccessible(true);
+    operatorImplGraphField.set(operatorTask, mockOperatorImplGraph);
```

Comment:
would be simpler to add a setter to the field and make it `package private`? Tests can leverage this in future and don't have to repeat this boiler plate of reflection code.